src/qibolab/backends.py

```diff
@@ -125,6 +125,13 @@ def execute_circuit(self, circuit, initial_state=None, nshots=1000):
                 "Hardware backend only supports circuits as initial states.",
             )
 
+        # This should be done in qibo side
+        # Temporary fix: overwrite the wire names
+        if not all(q in circuit.wire_names for q in self.platform.qubits):
+            circuit._wire_names = self.qubits
```

Comment:
> Do you actually need this fix?

No, if https://github.com/qiboteam/qibo/pull/1500 is applied, we can remove the temporary fix.

Currently, the circuit implementation in qibo uses `q{i}` as the default qubit names and does not support setting `wire_names` as `list[int]` through the setter. And there are minor issues where `wire_names` are not copied when copying the circuit. I included a temporary fix to avoid adding meaningless lines in the test files.